airbyte-integrations/connectors/source-paypal-transaction/source_paypal_transaction/manifest.yaml

```diff
@@ -150,63 +163,238 @@ definitions:
         field_name: as_of_time
         inject_into: request_parameter
     $parameters:
-      path: "balances"
+      path: "v1/reporting/balances"
+
+  #New Stream - List Product
+  #Paypal API only has V1 for this stream
+  list_products_stream:
+    type: DeclarativeStream
+    primary_key: id
+    name: "list_products"
+    retriever:
+      type: SimpleRetriever
+      record_selector:
+        $ref: "#/definitions/selector"
+      paginator:
+        type: DefaultPaginator
+        pagination_strategy:
+          type: PageIncrement
+          start_from_page: 1
+          page_size: 20
+        page_token_option:
+          type: RequestOption
+          inject_into: request_parameter
+          field_name: page
+        page_size_option:
+          inject_into: request_parameter
+          field_name: page_size
+          type: RequestOption
+      requester:
+        $ref: "#/definitions/requester"
+        http_method: GET
+        request_headers:
+          Content-Type: application/json
+    $parameters:
+      path: "v1/catalogs/products"
+      field_path: products
+
+  # New Stream - Show Product Details
+  #Paypal API only has V1 for this stream
+  #This can't be incremental as there is no time filtering. If you need to have the updates, you need to Append in the full_sync
+  # This stream works, however has some challenges with performance. Whith a big catalog it can take up to 3 hrs.
```

Comment:
yikes. [we do have a path forward to fetch requests concurrently in the low-code framework](https://docs.google.com/document/d/1YI-0Pud24eBswurq-J51BFwfKRHktBgJWFc2wtfQjL0/edit?usp=sharing). Unclear when it'll be prioritized

Reply:
I will take a look. But I don't think there is a current solution for that now. The customer will have to wait as this is an API limitation by itself.